### DISCUSSION TITLE
The proxy I shared from others has expired

### DISCUSSION THREAD

**Discussion by Tnze** [Tnze/go-mc#153] [General]
So I can't use Discord now. It's seems that GitHub Discussion can be used as an alternative.

**Comment by Tnze** [comment]
@maxsupermanhd 

Your negative Y is fixed.

**Reply by maxsupermanhd** [reply]
Thank you!

**Comment by patyhank** [comment]
How to read Chunk from the packet using level.Chunk?

**Comment by maxsupermanhd** [comment]
So I have few questions to new structure you have here, first how to check if palette has certain block?
And how do I loop over blocks now when `fromID` is not exported?
And why blocks were moved from data to level?

**Reply by Tnze** [reply]
You can't do so right now. Maybe you can show me that is a useful operation.

This is a new block package, and the old one is deleted because outdated. It's not moved.

**Reply by maxsupermanhd** [reply]
https://github.com/maxsupermanhd/WebChunk/blob/master/colors.go
For example here I handle assigned to blocks colors and I really really lack functionality to loop over block ID

**Reply by maxsupermanhd** [reply]
Why not just make `fromID` exported?

**Reply by Tnze** [reply]
That also allows users modified the map. So we need take care

**Reply by maxsupermanhd** [reply]
If they modify it they know what they are doing, why restrict? If someone trashes this map it's their fault

**Reply by Tnze** [reply]
I will support them as soon as I can. Including blocks or states iterations and chunk receiving for bot package.
But it takes times for writing and testing.

**Reply by Tnze** [reply]
The `FromID`, `ToStateID` and ` StateList` are public now as U wishes.

**Comment by bendgk** [comment]
@Tnze sad to hear your proxy expired, any way how I could chip in and help?

 * Do you know which VPN's work in China?
 * I would be willing to chip in 1-3 months' worth of VPN cost
 *  :)

**Reply by Tnze** [reply]
Thank you but I don't know :)
I don't really need this
